system/Config/BaseService.php

```diff
@@ -185,7 +185,7 @@ protected static function getSharedInstance(string $key, ...$params)
 			// Make sure $getShared is false
 			$params[] = false;
 
-			static::$instances[$key] = static::$key(...$params);
+			static::$instances[$key] = \Config\Services::$key(...$params);
```

Comment:
Okay, i left it there intentionally to prevent irritations about this later (someone thinking "oh wow, its `Services`, we just could call `static`" 😉).

Reply:
For a compromise, maybe we can do is do the import and add a comment like `// NOTE: To enable overrides by service providers, do not change to ``static`` calls`